include/wlr/types/wlr_client_buffer.h

```diff
@@ -13,9 +13,25 @@
 #include <wayland-server-core.h>
 #include <wlr/types/wlr_buffer.h>
 
+struct wlr_renderer;
+struct wlr_client_buffer;
+
+struct wlr_client_buffer_impl {
+	bool (*is_instance)(struct wl_resource *resource);
+	bool (*get_buffer_size)(struct wl_resource *resource,
```

Comment:
I'd be nice if we could avoid implementing this function, and instead import the buffer before we need to know the size in `wl_surface.commit`'s handler.

Reply:
Maybe we could wrap it in the `init`? The function is also uses in [`wlr_surface`](https://github.com/swaywm/wlroots/blob/master/types/wlr_surface.c#L176), do you think we can do without by storing the size in the `wlr_client_buffer` ?

Reply:
The size is in `wlr_client_buffer.base.{width,height}`.

Reply:
(made the edit while you were commenting :p )